paddle/fluid/framework/shape_inference.h

```diff
@@ -25,7 +25,10 @@ limitations under the License. */
 namespace paddle {
 namespace framework {
 
+class OperatorBase;
+
 using InferShapeVarPtr = boost::variant<VarDesc *, Variable *>;
+using InferShapeOpPtr = boost::variant<const OpDesc *, const OperatorBase *>;
```

Comment:
why make it variant? Seems only OpDesc is used?

Reply:
Yes. But `InferShapeContext` is an abstract class hidden in source file. Making it as a `variant` to make it more general.